lib/reservation.rb

```diff
@@ -0,0 +1,42 @@
+require_relative "room"
+require_relative "hotel"
+
+module HotelSystem
+  class Reservation
+    attr_reader :room_number, :id
+    attr_accessor :date_range, :room, :nightly_rate
+
+    def initialize(id:, room: nil, room_number: nil, date_range:, nightly_rate: 200)
+      @id = id
+      if room
+        valid_room(room)
+        @room = room
+        @room_number = room.room_number
+      elsif room_number
+        valid_room_number(room_number)
+        @room_number = room_number
```

Comment:
This is probably more flexibility that I would like for this class. If it needs an instance of `Room` it should require it, and if it doesn't need it then it shouldn't ask for it, and if you have a good reason to accept both you should leave a big comment here for whoever comes next.